cigaradvisor/scripts/scripts.js

```diff
@@ -248,21 +248,27 @@ export function getRelativePath(path) {
   return relPath;
 }
 
-let articleIndexData;
-
+const articleIndexData = [];
 /**
  * Returns all the posts in the posts index.
  *
  * @return {Promise<Array[Object]>}
  */
-export async function loadPosts() {
-  if (!articleIndexData) {
-    const resp = await fetch(ARTICLE_INDEX_PATH);
+export async function loadPosts(path = ARTICLE_INDEX_PATH, flag = false) {
+  if (articleIndexData.length === 0 || flag) {
+    const resp = await fetch(path);
     let jsonData = '';
     if (resp.ok) {
       jsonData = await resp.json();
     }
-    articleIndexData = jsonData.data;
+    jsonData.data.forEach((a) => {
+      articleIndexData.push({ ...a });
+    });
+    // If there are more articles to load, load them
+    if ((jsonData.total - jsonData.offset) > jsonData.limit) {
```

Comment:
We shouldn't load _all of the posts_, unless the caller actually wants _all of the posts_. So we need to probably reassess how this entire post logic works.

Reply:
hmm...I am trying to think about a case where we don't have to get all the posts.

Reply:
If we're going to load all of the posts, is there a reason we need to recurse into this method, rather than just loop?